dev/sg/sg_ci.go

```diff
@@ -220,16 +226,39 @@ sg ci build --help
 				return err
 			}
 
-			previewCmd := usershell.Command(cmd.Context, "go run ./enterprise/dev/ci/gen-pipeline.go -preview").
-				Env(map[string]string{
-					"BUILDKITE_BRANCH":  target.target, // this must be a branch
-					"BUILDKITE_MESSAGE": message,
-				})
-			out, err := root.Run(previewCmd).String()
-			if err != nil {
-				return err
+			var previewCmd *sgrun.Command
+			env := map[string]string{
+				"BUILDKITE_BRANCH":  target.target, // this must be a branch
+				"BUILDKITE_MESSAGE": message,
+			}
+			switch cmd.String("format") {
+			case "markdown":
+				previewCmd = usershell.Command(cmd.Context, "go run ./enterprise/dev/ci/gen-pipeline.go -preview").
+					Env(env)
+				out, err := root.Run(previewCmd).String()
+				if err != nil {
+					return err
+				}
+				return std.Out.WriteMarkdown(out)
+			case "json":
+				previewCmd = usershell.Command(cmd.Context, "go run ./enterprise/dev/ci/gen-pipeline.go").
+					Env(env)
+				out, err := root.Run(previewCmd).String()
+				if err != nil {
+					return err
+				}
+				return std.Out.WriteCode("json", out)
+			case "yaml":
+				previewCmd = usershell.Command(cmd.Context, "go run ./enterprise/dev/ci/gen-pipeline.go -yaml").
+					Env(env)
+				out, err := root.Run(previewCmd).String()
+				if err != nil {
+					return err
+				}
+				return std.Out.WriteCode("yaml", out)
+			default:
+				return flag.ErrHelp
```

Comment:
nit - I always find returning `flag.ErrHelp` super unhelpful, because it tells you nothing about what went wrong and sometimes just dumps a wall of text that hides the output 😆 

```suggestion
				return errors.Newf("unsupported format type: %q", cmd.String("format")
```